t/break-undo.vim

```diff
@@ -0,0 +1,125 @@
+" NB: MacVim defines several key mappings such as <D-v> by default.
+" The key mappings are defined from the core, not from any runtime file.
+" So that the key mappings are always defined even if Vim is invoked by
+" "vim -u NONE" etc.  Remove the kay mappings to ensure that there is no key
+" mappings, because some tests in this file assume such state.
+imapclear
+cmapclear
+
+let g:smartinput_break_undo = 1
+runtime! plugin/smartinput.vim
+
+call vspec#hint({'scope': 'smartinput#scope()', 'sid': 'smartinput#sid()'})
+set backspace=indent,eol,start
+filetype plugin indent on
+syntax enable
+
+describe 'g:smartinput_break_undo'
+  before
+    SaveContext
+    call smartinput#clear_rules()
+    new
+
+    function! b:.test_keys_undo(name, pre_input, table)
+      let n = a:name
+      for i in range(len(a:table))
+        let [input, expects] = a:table[i]
+        normal! gg0D
+        call feedkeys('a' . a:pre_input, 'tx')
+        call feedkeys(input, 'tx')
+        for j in range(len(expects))
+          let expected = expects[j]
+          Expect [n, i, j, getline(1, line('$'))] ==# [n, i, j, [expected]]
+          normal! u
+        endfor
+      endfor
+    endfunction
+
+    function! b:.test_no_break_undo()
+      " define with "i_CTRL-G_U"
+      call smartinput#define_default_rules()
+      Expect string(Ref('s:available_nrules')) =~# '<C-g>U'
```

Comment:
Do not check this variable.  This is an implementation detail which might be changed without changing behavior.